File: diverta2019/b.cpp
#include <string>
#include <iostream>

using namespace std;

int main() {
    int R, G, B, N;
    cin >> R >> G >> B >> N;

    int ans = 0;
    for (int r=0; r<3001; r++) {
        int balls = R*r;
        if (N < balls) {
            break;
        }
        for (int g=0; g<3001; g++) {
            balls = R*r + G*g;
            if (N < balls) {
                break;
            }
            int left = N - balls;
            if (left%B == 0) {
                ans += 1;
            }
        }
    }

    cout << ans << endl;
}

File: diverta2019/b.py
R, G, B, N = map(int, input().split())

ans = 0
rsum = 0
for r in range(0, N+1):
    if N < rsum:
        break
    rgsum = rsum
    for g in range(0, N+1):
        if N < rgsum:
            break
        left = N - rgsum 
        if left % B == 0:
            ans += 1
        rgsum += G
    rsum += R

print(ans)

